library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/processor/RecordedDataProcessor.kt

```diff
@@ -165,28 +157,7 @@ internal class RecordedDataProcessor(
         }
     }
 
-    private fun executeRunnable(runnable: Runnable) {
-        @Suppress("SwallowedException", "TooGenericExceptionCaught")
-        try {
-            executorService.submit(runnable)
-        } catch (e: RejectedExecutionException) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-        } catch (e: NullPointerException) {
-            // TODO: RUMM-2397 Add the proper logs here once the sdkLogger will be added
-            // the task will never be null so normally this exception should not be triggered.
-            // In any case we will add a log here later.
-        }
-    }
-
-    private fun buildRunnable(
-        runnableFactory: (
-            timestamp: Long,
-            newContext: SessionReplayRumContext,
-            prevRumContext: SessionReplayRumContext
-        ) -> Runnable
-    ): Runnable? {
-        val rumContextData = rumContextDataHandler.createRumContextData() ?: return null
-
+    private fun updateViewSent(rumContext: SessionReplayRumContext) {
```

Comment:
Looks good but we missed something. This method was before called from `MainThread` intentionally because we want to update the `Rum` context in time to mark the `has_replay` on the `RumViewEvent` in progress. Right now all these methods are  called in the `WorkerThread` so we will have to move this call from here and mainly the `recordCallback` somewhere else. I'd say a good moment will be when we put the item in the queue, probably in the `QueueHandler`

Reply:
When we initially put the item in the queue we don't know for sure that the traversal will succeed and we will have nodes. What about putting it right after the end of the traversal where we set the nodes, before the call to awakening the queue?

Reply:
As we discussed, I've moved recordCallback inside the writer